adbc.h

```diff
@@ -1380,6 +1827,67 @@ AdbcStatusCode AdbcConnectionGetTableTypes(struct AdbcConnection* connection,
                                            struct ArrowArrayStream* out,
                                            struct AdbcError* error);
 
+/// \brief Get a view of types supported by the database.
+///
+/// The result is an Arrow schema with one field per type.  The field name is
+/// the database's name for the type, and the field type is the driver's
+/// mapping of the database type to an Arrow type.
```

Comment:
How does it work for e.g. timestamps or fixed-size strings? Should the database return all supported timezones? One of them? Same question for decimals.

Reply:
Perhaps an unsatisfactory answer, but this is based on the database's idea of types, so not every Arrow type is necessarily represented. However, I'll dig up concrete examples of what happens with decimal/timestamp types in databases.

(Timezone handling, though, is a mess, and I'll point out that e.g. Postgres and SQL Server just have naive timestamps + UTC timestamps, with the timezone being purely a display concept that is set on the fly)

Reply:
Is this expected to be the same as `AdbcConnectionGetTableTypes` except for having the arrow mapping? Is there a reason to have both?

Reply:
Ah, this is about data types, not table types. That said: I'm inclined to drop this too (practically, you only care about the types in the context of a table, in which case GetTableSchema should be sufficient)